File: source/ProductsInventory.DB/Domain/Category.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ProductsInventory.DB.Domain
{
    public class Category : EntityBase
    {
        [Key]
        public Guid CategoryId { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string CategoryCode { get; set; }
        [Required]
        public bool IsActive { get; set; }

        public Guid UserId { get; set; }
        public virtual List<Product> Products { get; set; }
    }
}

File: client/src/app/shared/models/category.ts
import { Product } from './product';
export class Category {
    categoryId: string;
    name: string;
    categoryCode: string;
    isActive: boolean;
    dateCreated: Date;
    dateLastModified: Date;   
    createdBy: string;
    lastUpdatedBy: string;

    products: Product[];
}

